torch/csrc/jit/codegen/cuda/contiguity.h

```diff
@@ -65,11 +75,27 @@ class ContigIDs : public OptInDispatch {
 
   void handle(Merge* merge) override;
 
+  IterDomain* getCAIndexConcreteId(IterDomain* id) const;
+
+  //! True if an ID is indexable.
+  //! E.g., a merged domain with broadcast may not be indexable when
+  //! its corresponding reference tensor has non-broadcast domains.
+  bool isIndexable(IterDomain* id) const;
```

Comment:
If we didn't use a reference and instead had direct indexing on the index map, would this still be an issue? or we could confidently just use the index in the index map?

Reply:
I think in either case we would need to check if an IterDomain is indexable, which is equivalent to saying a mapped index for the IterDomain is found in the index map.